vignettes/epipredict.Rmd

```diff
@@ -1,470 +1,565 @@
+However, geo-pooling can only be used if values are comparable in meaning and scale across geographies or can be made comparable, for example by normalization.
+
+If we wanted to build a geo-aware model, such as a linear regression with a different intercept for each geography, we would need to build a [custom workflow](custom_epiworkflows) with geography as a factor.
+
+# Anatomy of a canned forecaster
+## Code object
+Let's dissect the forecaster we trained back on the [landing
+page](../index.html#motivating-example):
+
+```{r make-four-forecasts, warning=FALSE}
+four_week_ahead <- arx_forecaster(
+  covid_case_death_rates |> filter(time_value <= forecast_date),
+  outcome = "death_rate",
+  predictors = c("case_rate", "death_rate"),
+  args_list = arx_args_list(
+    lags = list(c(0, 1, 2, 3, 7, 14), c(0, 7, 14)),
+    ahead = 4 * 7,
+    quantile_levels = c(0.1, 0.25, 0.5, 0.75, 0.9)
+  )
+)
 ```
 
-At predict time, we add this object onto the `epi_workflow` and call `forecast()`
+`four_week_ahead` has two components: an `epi_workflow`, and a table of
+`predictions`.
+The table of predictions is a simple tibble,
 
-```{r, warning=FALSE}
-ewf %>%
-  add_frosting(f) %>%
-  forecast()
+```{r show_predictions}
+four_week_ahead$predictions
 ```
 
-The above `get_test_data()` function examines the recipe and ensures that enough
-test data is available to create the necessary lags and produce a prediction
-for the desired future time point (after the end of the training data). This mimics
-what would happen if `jhu` contained the most recent available historical data and
-we wanted to actually predict the future. We could have instead used any test data
-that contained the necessary predictors.
+where `.pred` gives the point/median prediction, and `.pred_distn` is a
+`dist_quantiles()` object representing a distribution through various quantile
+levels.
+The `[6]` in the name refers to the number of quantiles that have been
+explicitly created[^4].
+By default, `.pred_distn` covers a 90% prediction interval, reporting the 5% and 95% quantiles.
+
+The `epi_workflow` is a significantly more complicated object, extending a
+`workflows::workflow()`  to include post-processing steps:
 
+```{r show_workflow}
+four_week_ahead$epi_workflow
+```
 
-## Conclusion
+An `epi_workflow()` consists of 3 parts:
+
+- `Preprocessor`: a collection of steps that transform the data to be ready for
+      modelling. Steps can be custom, as are those included in this package, 
+      or [be defined in `{recipes}`](https://recipes.tidymodels.org/reference/index.html).
+      `four_week_ahead` has 5 steps; you can inspect them more closely by
+      running `hardhat::extract_recipe(four_week_ahead$epi_workflow)`.[^6]
+- `Model`: the actual model that does the fitting, given by a
+  `parsnip::model_spec`. `four_week_ahead` uses the default of
+  `parsnip::linear_reg()`, which is a `{parsnip}` wrapper for
+  `stats::lm()`. You can inspect the model more closely by running
```

Comment:
it only wraps `lm()` because `engine = "lm"` by default. There are many other
engines that also estimate linear models.